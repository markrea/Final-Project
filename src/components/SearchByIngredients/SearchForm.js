import React from "react";
import "../../styles/Search.css";
import Select from "react-select";
import Proptypes from "prop-types";

const SearchForm = ({
  handleSubmit,
  handleChange,
  handleSelect,
  query,
  diet,
  setIntolerances,
}) => {
  const options = [
    { value: "dairy", label: "Dairy" },
    { value: "egg", label: "Egg" },
    { value: "gluten", label: "Gluten" },
    { value: "peanut", label: "Peanut" },
    { value: "seafood", label: "Seafood" },
    { value: "sesame", label: "Sesame" },
    { value: "shellfish", label: "Shellfish" },
    { value: "soy", label: "Soy" },
    { value: "treenut", label: "Tree Nut" },
  ];

  return (
    <form
      type="text"
      className="input-form"
      onSubmit={handleSubmit}
      data-testid="searchForm"
    >
      <fieldset>
        <select
          id="diet"
          name="diet"
          size="1"
          value={diet}
          placeholder="Select Diet"
          onChange={handleSelect}
        >
          <option defaultValue="null">Select Diet</option>
          <option value="vegetarian">Vegetarian</option>
          <option value="vegan">Vegan</option>
          <option value="glutenfree">Gluten Free</option>
          <option value="pescatarian">Pescastarian</option>
        </select>
        <Select
          isMulti
          autoFocus
          name="colors"
          options={options}
          className="select-container"
          classNamePrefix="select"
          placeholder="Select intolerances"
          onChange={setIntolerances}
          theme={(theme) => ({
            ...theme,
            borderRadius: "10px",
            backgroundColor: "rgba(196, 208, 162, 1)",
          })}
        />
        <br />

        <input
          id="search"
          name="recipeName"
          type="text"
          placeholder="Enter ingredients"
          onChange={handleChange}
          value={query}
          data-testid="input-id"
        />
        <button type="submit" className="submit">
          Go!
        </button>
        <br />
      </fieldset>
    </form>
  );
};

SearchForm.propTypes = {
  handleSubmit: Proptypes.func.isRequired,
  handleChange: Proptypes.func.isRequired,
  handleSelect: Proptypes.func.isRequired,
  query: Proptypes.string.isRequired,
  diet: Proptypes.string.isRequired,
  setIntolerances: Proptypes.string.isRequired,
};

export default SearchForm;
