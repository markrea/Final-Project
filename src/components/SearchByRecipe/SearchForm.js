import React from "react";
import "../../styles/Search.css";

const SearchForm = ({ handleSubmit, handleChange, query }) => {
  return (
    <form
      type="text"
      className="input-form"
      onSubmit={handleSubmit}
      data-testid="searchForm"
    >
      <fieldset>
        <input
          id="search"
          name="recipeName"
          type="text"
          placeholder="Enter ingredients"
          onChange={handleChange}
          value={query}
          data-testid="input-id"
        />
        <button type="submit" className="submit">
          Go!
        </button>
      </fieldset>
    </form>
  );
};

export default SearchForm;
