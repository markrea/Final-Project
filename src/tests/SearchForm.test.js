import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SearchForm from "../components/SearchByIngredients/SearchForm";
import "@testing-library/jest-dom/extend-expect";

const props = {
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  query: "random",
};

describe("SearchForm", () => {
  it("component renders with correct props", () => {
    const { asFragment } = render(<SearchForm {...props} />);
    expect(asFragment).toMatchSnapshot();
  });
  it("renders a form", () => {
    const { getByTestId } = render(<SearchForm {...props} />);
    const form = getByTestId("searchForm");
    expect(form).toBeInTheDocument();
  });
  it("renders an input for the form", () => {
    const { getByTestId } = render(<SearchForm {...props} />);
    const input = getByTestId("input-id");
    expect(input).toBeInTheDocument();
  });
  it("renders a button", () => {
    const { getByRole } = render(<SearchForm {...props} />);
    const btn = getByRole("button");
    expect(btn).toBeInTheDocument();
    expect(btn).toHaveTextContent("Go");
  });
  it("changes the state when input is changed", () => {
    const { getByTestId } = render(<SearchForm {...props} />);
    const { handleChange } = props;
    const input = getByTestId("input-id");
    fireEvent.change(input, { target: { value: "randomQuery" } });
    expect(handleChange).toHaveBeenCalled();
  });
  it("calls handleSearch on submission of form", () => {
    const { getByTestId } = render(<SearchForm {...props} />);
    const { handleSubmit } = props;
    const form = getByTestId("searchForm");
    fireEvent.submit(form);
    expect(handleSubmit).toHaveBeenCalled();
  });
});
