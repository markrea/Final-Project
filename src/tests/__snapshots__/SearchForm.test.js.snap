// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders with correct props 1`] = `[Function]`;
